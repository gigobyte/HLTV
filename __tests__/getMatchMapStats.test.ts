import HLTV from '../src/'

function sleep(ms: number): Promise<void> {
  return new Promise((resolve) => setTimeout(resolve, ms))
}

const FULL_INFO_NEWEST = 116688
const FULL_INFO = 79440
const RATING_1 = 14783

test('getMatchMapStats', async () => {
  await sleep(3000)
  expect(
    await HLTV.getMatchMapStats({ id: FULL_INFO_NEWEST })
  ).toMatchSnapshot()
  expect(await HLTV.getMatchMapStats({ id: FULL_INFO })).toMatchSnapshot()
  await sleep(3000)
  expect(await HLTV.getMatchMapStats({ id: RATING_1 })).toMatchSnapshot()
  await sleep(3000)
}, 30000)
