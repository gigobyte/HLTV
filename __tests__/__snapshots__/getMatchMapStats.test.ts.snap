// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMatchMapStats 1`] = `
Object {
  "date": 1616065500000,
  "event": Object {
    "id": 5785,
    "name": "Snow Sweet Snow 3 Regionals",
  },
  "id": 116688,
  "map": "de_vertigo",
  "matchId": 2347210,
  "overview": Object {
    "bestRating2": Object {
      "id": 16410,
      "name": "Duplicate",
      "value": 1.53,
    },
    "clutchesWon": Object {
      "team1": 1,
      "team2": 1,
    },
    "firstKills": Object {
      "team1": 18,
      "team2": 8,
    },
    "mostAWPKills": Object {
      "id": 12379,
      "name": "Impulse",
      "value": 19,
    },
    "mostAssists": Object {
      "id": 3730,
      "name": "HOLMES",
      "value": 7,
    },
    "mostDamage": Object {
      "id": 12379,
      "name": "Impulse",
      "value": 108,
    },
    "mostFirstKills": Object {
      "id": 13466,
      "name": "dennyslaw",
      "value": 8,
    },
    "mostKills": Object {
      "id": 12379,
      "name": "Impulse",
      "value": 30,
    },
    "rating": Object {
      "team1": 1.27,
      "team2": 0.85,
    },
  },
  "performanceOverview": Object {
    "team1": Object {
      "assists": 19,
      "deaths": 76,
      "kills": 98,
    },
    "team2": Object {
      "assists": 14,
      "deaths": 98,
      "kills": 76,
    },
  },
  "playerStats": Object {
    "team1": Array [
      Object {
        "ADR": 107.9,
        "KAST": 76.9,
        "assists": 8,
        "deaths": 15,
        "deathsPerRound": 0.58,
        "firstKillsDifference": -3,
        "flashAssists": 1,
        "hsKills": 11,
        "impact": 1.56,
        "killDeathsDifference": 8,
        "kills": 23,
        "killsPerRound": 0.88,
        "player": Object {
          "id": 16410,
          "name": "Duplicate",
        },
        "rating1": 1.53,
      },
      Object {
        "ADR": 71,
        "KAST": 80.8,
        "assists": 4,
        "deaths": 15,
        "deathsPerRound": 0.58,
        "firstKillsDifference": 6,
        "flashAssists": 3,
        "hsKills": 13,
        "impact": 1.75,
        "killDeathsDifference": 8,
        "kills": 23,
        "killsPerRound": 0.88,
        "player": Object {
          "id": 13466,
          "name": "dennyslaw",
        },
        "rating1": 1.4,
      },
      Object {
        "ADR": 73.5,
        "KAST": 80.8,
        "assists": 6,
        "deaths": 13,
        "deathsPerRound": 0.5,
        "firstKillsDifference": 1,
        "flashAssists": 1,
        "hsKills": 7,
        "impact": 0.95,
        "killDeathsDifference": 6,
        "kills": 19,
        "killsPerRound": 0.73,
        "player": Object {
          "id": 17145,
          "name": "Rainwaker",
        },
        "rating1": 1.21,
      },
      Object {
        "ADR": 65,
        "KAST": 80.8,
        "assists": 1,
        "deaths": 15,
        "deathsPerRound": 0.58,
        "firstKillsDifference": 3,
        "flashAssists": 0,
        "hsKills": 10,
        "impact": 0.76,
        "killDeathsDifference": 2,
        "kills": 17,
        "killsPerRound": 0.65,
        "player": Object {
          "id": 18697,
          "name": "Oxygen",
        },
        "rating1": 1.09,
      },
      Object {
        "ADR": 75.8,
        "KAST": 73.1,
        "assists": 7,
        "deaths": 18,
        "deathsPerRound": 0.69,
        "firstKillsDifference": 3,
        "flashAssists": 2,
        "hsKills": 9,
        "impact": 1.12,
        "killDeathsDifference": -2,
        "kills": 16,
        "killsPerRound": 0.62,
        "player": Object {
          "id": 11343,
          "name": "SHiPZ",
        },
        "rating1": 1.09,
      },
    ],
    "team2": Array [
      Object {
        "ADR": 108,
        "KAST": 69.2,
        "assists": 5,
        "deaths": 17,
        "deathsPerRound": 0.65,
        "firstKillsDifference": -1,
        "flashAssists": 4,
        "hsKills": 7,
        "impact": 1.58,
        "killDeathsDifference": 13,
        "kills": 30,
        "killsPerRound": 1.15,
        "player": Object {
          "id": 12379,
          "name": "Impulse",
        },
        "rating1": 1.5,
      },
      Object {
        "ADR": 53.2,
        "KAST": 61.5,
        "assists": 2,
        "deaths": 20,
        "deathsPerRound": 0.77,
        "firstKillsDifference": -3,
        "flashAssists": 0,
        "hsKills": 9,
        "impact": 0.67,
        "killDeathsDifference": -5,
        "kills": 15,
        "killsPerRound": 0.58,
        "player": Object {
          "id": 3732,
          "name": "DiMKE",
        },
        "rating1": 0.76,
      },
      Object {
        "ADR": 67.1,
        "KAST": 53.8,
        "assists": 3,
        "deaths": 21,
        "deathsPerRound": 0.81,
        "firstKillsDifference": -1,
        "flashAssists": 0,
        "hsKills": 6,
        "impact": 0.58,
        "killDeathsDifference": -8,
        "kills": 13,
        "killsPerRound": 0.5,
        "player": Object {
          "id": 5351,
          "name": "aVN",
        },
        "rating1": 0.71,
      },
      Object {
        "ADR": 51.5,
        "KAST": 61.5,
        "assists": 5,
        "deaths": 19,
        "deathsPerRound": 0.73,
        "firstKillsDifference": -1,
        "flashAssists": 4,
        "hsKills": 7,
        "impact": 0.53,
        "killDeathsDifference": -8,
        "kills": 11,
        "killsPerRound": 0.42,
        "player": Object {
          "id": 4218,
          "name": "Dav",
        },
        "rating1": 0.68,
      },
      Object {
        "ADR": 53.5,
        "KAST": 61.5,
        "assists": 9,
        "deaths": 21,
        "deathsPerRound": 0.81,
        "firstKillsDifference": -4,
        "flashAssists": 2,
        "hsKills": 5,
        "impact": 0.42,
        "killDeathsDifference": -14,
        "kills": 7,
        "killsPerRound": 0.27,
        "player": Object {
          "id": 3730,
          "name": "HOLMES",
        },
        "rating1": 0.58,
      },
    ],
  },
  "result": Object {
    "halfResults": Array [
      Object {
        "team1Rounds": 11,
        "team2Rounds": 4,
      },
      Object {
        "team1Rounds": 5,
        "team2Rounds": 6,
      },
    ],
    "team1TotalRounds": 16,
    "team2TotalRounds": 10,
  },
  "roundHistory": Array [
    Object {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "1-0",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "2-0",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "3-0",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "bomb_exploded",
      "score": "4-0",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "bomb_exploded",
      "score": "5-0",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "6-0",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "7-0",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "bomb_exploded",
      "score": "8-0",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "9-0",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "ct_win",
      "score": "9-1",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "ct_win",
      "score": "9-2",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "10-2",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "bomb_defused",
      "score": "10-3",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "ct_win",
      "score": "10-4",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "11-4",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 11001,
      "outcome": "t_win",
      "score": "11-5",
      "tTeam": 10386,
    },
    Object {
      "ctTeam": 10386,
      "outcome": "t_win",
      "score": "11-6",
      "tTeam": 11001,
    },
    Object {
      "ctTeam": 10386,
      "outcome": "t_win",
      "score": "11-7",
      "tTeam": 11001,
    },
    Object {
      "ctTeam": 10386,
      "outcome": "t_win",
      "score": "11-8",
      "tTeam": 11001,
    },
    Object {
      "ctTeam": 10386,
      "outcome": "ct_win",
      "score": "12-8",
      "tTeam": 11001,
    },
    Object {
      "ctTeam": 10386,
      "outcome": "stopwatch",
      "score": "13-8",
      "tTeam": 11001,
    },
    Object {
      "ctTeam": 10386,
      "outcome": "bomb_defused",
      "score": "14-8",
      "tTeam": 11001,
    },
    Object {
      "ctTeam": 10386,
      "outcome": "bomb_exploded",
      "score": "14-9",
      "tTeam": 11001,
    },
    Object {
      "ctTeam": 10386,
      "outcome": "t_win",
      "score": "14-10",
      "tTeam": 11001,
    },
    Object {
      "ctTeam": 10386,
      "outcome": "ct_win",
      "score": "15-10",
      "tTeam": 11001,
    },
    Object {
      "ctTeam": 10386,
      "outcome": "ct_win",
      "score": "16-10",
      "tTeam": 11001,
    },
  ],
  "team1": Object {
    "id": 10386,
    "name": "SKADE",
  },
  "team2": Object {
    "id": 11001,
    "name": "ONYX",
  },
}
`;

exports[`getMatchMapStats 2`] = `
Object {
  "date": 1544901600000,
  "event": Object {
    "id": 4205,
    "name": "BLAST Pro Series Lisbon 2018",
  },
  "id": 79440,
  "map": "de_overpass",
  "matchId": 2329728,
  "overview": Object {
    "bestRating2": Object {
      "id": 7998,
      "name": "s1mple",
      "value": 1.61,
    },
    "clutchesWon": Object {
      "team1": 0,
      "team2": 3,
    },
    "firstKills": Object {
      "team1": 10,
      "team2": 13,
    },
    "mostAWPKills": Object {
      "id": 7998,
      "name": "s1mple",
      "value": 9,
    },
    "mostAssists": Object {
      "id": 484,
      "name": "Zeus",
      "value": 9,
    },
    "mostDamage": Object {
      "id": 7398,
      "name": "dupreeh",
      "value": 99.9,
    },
    "mostFirstKills": Object {
      "id": 8918,
      "name": "electronic",
      "value": 5,
    },
    "mostKills": Object {
      "id": 7398,
      "name": "dupreeh",
      "value": 23,
    },
    "rating": Object {
      "team1": 0.81,
      "team2": 1.28,
    },
  },
  "performanceOverview": Object {
    "team1": Object {
      "assists": 10,
      "deaths": 90,
      "kills": 65,
    },
    "team2": Object {
      "assists": 17,
      "deaths": 65,
      "kills": 90,
    },
  },
  "playerStats": Object {
    "team1": Array [
      Object {
        "ADR": 99.9,
        "KAST": 78.3,
        "assists": 1,
        "deaths": 19,
        "deathsPerRound": 0.83,
        "firstKillsDifference": 1,
        "flashAssists": 0,
        "hsKills": 9,
        "impact": 1.51,
        "killDeathsDifference": 4,
        "kills": 23,
        "killsPerRound": 1,
        "player": Object {
          "id": 7398,
          "name": "dupreeh",
        },
        "rating1": 1.32,
      },
      Object {
        "ADR": 78.7,
        "KAST": 56.5,
        "assists": 4,
        "deaths": 17,
        "deathsPerRound": 0.74,
        "firstKillsDifference": 0,
        "flashAssists": 1,
        "hsKills": 9,
        "impact": 0.87,
        "killDeathsDifference": -3,
        "kills": 14,
        "killsPerRound": 0.61,
        "player": Object {
          "id": 9032,
          "name": "Magisk",
        },
        "rating1": 0.89,
      },
      Object {
        "ADR": 70,
        "KAST": 60.9,
        "assists": 3,
        "deaths": 19,
        "deathsPerRound": 0.83,
        "firstKillsDifference": 0,
        "flashAssists": 1,
        "hsKills": 4,
        "impact": 0.56,
        "killDeathsDifference": -7,
        "kills": 12,
        "killsPerRound": 0.52,
        "player": Object {
          "id": 7592,
          "name": "device",
        },
        "rating1": 0.72,
      },
      Object {
        "ADR": 50.3,
        "KAST": 52.2,
        "assists": 3,
        "deaths": 16,
        "deathsPerRound": 0.7,
        "firstKillsDifference": -3,
        "flashAssists": 1,
        "hsKills": 4,
        "impact": 0.34,
        "killDeathsDifference": -7,
        "kills": 9,
        "killsPerRound": 0.39,
        "player": Object {
          "id": 7412,
          "name": "gla1ve",
        },
        "rating1": 0.56,
      },
      Object {
        "ADR": 54.2,
        "KAST": 52.2,
        "assists": 2,
        "deaths": 19,
        "deathsPerRound": 0.83,
        "firstKillsDifference": -1,
        "flashAssists": 0,
        "hsKills": 1,
        "impact": 0.4,
        "killDeathsDifference": -12,
        "kills": 7,
        "killsPerRound": 0.3,
        "player": Object {
          "id": 4954,
          "name": "Xyp9x",
        },
        "rating1": 0.53,
      },
    ],
    "team2": Array [
      Object {
        "ADR": 98.6,
        "KAST": 78.3,
        "assists": 1,
        "deaths": 11,
        "deathsPerRound": 0.48,
        "firstKillsDifference": 1,
        "flashAssists": 0,
        "hsKills": 12,
        "impact": 1.8,
        "killDeathsDifference": 12,
        "kills": 23,
        "killsPerRound": 1,
        "player": Object {
          "id": 7998,
          "name": "s1mple",
        },
        "rating1": 1.61,
      },
      Object {
        "ADR": 93,
        "KAST": 73.9,
        "assists": 4,
        "deaths": 14,
        "deathsPerRound": 0.61,
        "firstKillsDifference": 2,
        "flashAssists": 1,
        "hsKills": 11,
        "impact": 1.8,
        "killDeathsDifference": 7,
        "kills": 21,
        "killsPerRound": 0.91,
        "player": Object {
          "id": 8918,
          "name": "electronic",
        },
        "rating1": 1.48,
      },
      Object {
        "ADR": 82.4,
        "KAST": 91.3,
        "assists": 2,
        "deaths": 12,
        "deathsPerRound": 0.52,
        "firstKillsDifference": 3,
        "flashAssists": 0,
        "hsKills": 8,
        "impact": 1.16,
        "killDeathsDifference": 8,
        "kills": 20,
        "killsPerRound": 0.87,
        "player": Object {
          "id": 483,
          "name": "Edward",
        },
        "rating1": 1.41,
      },
      Object {
        "ADR": 68.3,
        "KAST": 60.9,
        "assists": 4,
        "deaths": 13,
        "deathsPerRound": 0.57,
        "firstKillsDifference": -2,
        "flashAssists": 2,
        "hsKills": 8,
        "impact": 0.91,
        "killDeathsDifference": 2,
        "kills": 15,
        "killsPerRound": 0.65,
        "player": Object {
          "id": 7594,
          "name": "flamie",
        },
        "rating1": 1,
      },
      Object {
        "ADR": 74,
        "KAST": 69.6,
        "assists": 11,
        "deaths": 15,
        "deathsPerRound": 0.65,
        "firstKillsDifference": -1,
        "flashAssists": 2,
        "hsKills": 4,
        "impact": 0.65,
        "killDeathsDifference": -4,
        "kills": 11,
        "killsPerRound": 0.48,
        "player": Object {
          "id": 484,
          "name": "Zeus",
        },
        "rating1": 0.91,
      },
    ],
  },
  "result": Object {
    "halfResults": Array [
      Object {
        "team1Rounds": 4,
        "team2Rounds": 11,
      },
      Object {
        "team1Rounds": 3,
        "team2Rounds": 5,
      },
    ],
    "team1TotalRounds": 7,
    "team2TotalRounds": 16,
  },
  "roundHistory": Array [
    Object {
      "ctTeam": 6665,
      "outcome": "bomb_defused",
      "score": "1-0",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "t_win",
      "score": "1-1",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "t_win",
      "score": "1-2",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "t_win",
      "score": "1-3",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "bomb_exploded",
      "score": "1-4",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "t_win",
      "score": "1-5",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "stopwatch",
      "score": "2-5",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "t_win",
      "score": "2-6",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "t_win",
      "score": "2-7",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "t_win",
      "score": "2-8",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "ct_win",
      "score": "3-8",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "stopwatch",
      "score": "4-8",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "t_win",
      "score": "4-9",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "t_win",
      "score": "4-10",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "t_win",
      "score": "4-11",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 6665,
      "outcome": "ct_win",
      "score": "4-12",
      "tTeam": 4608,
    },
    Object {
      "ctTeam": 4608,
      "outcome": "ct_win",
      "score": "4-13",
      "tTeam": 6665,
    },
    Object {
      "ctTeam": 4608,
      "outcome": "bomb_exploded",
      "score": "5-13",
      "tTeam": 6665,
    },
    Object {
      "ctTeam": 4608,
      "outcome": "bomb_exploded",
      "score": "6-13",
      "tTeam": 6665,
    },
    Object {
      "ctTeam": 4608,
      "outcome": "ct_win",
      "score": "6-14",
      "tTeam": 6665,
    },
    Object {
      "ctTeam": 4608,
      "outcome": "bomb_exploded",
      "score": "7-14",
      "tTeam": 6665,
    },
    Object {
      "ctTeam": 4608,
      "outcome": "bomb_defused",
      "score": "7-15",
      "tTeam": 6665,
    },
    Object {
      "ctTeam": 4608,
      "outcome": "ct_win",
      "score": "7-16",
      "tTeam": 6665,
    },
  ],
  "team1": Object {
    "id": 6665,
    "name": "Astralis",
  },
  "team2": Object {
    "id": 4608,
    "name": "Natus Vincere",
  },
}
`;
